### PR TITLE
Add workarounds to allow glibmm < 2.64 and gtkmm < 3.24

### PR DESCRIPTION
**Pull request by ForeverRainbow**
This adds workarounds to allow compiling with older versions of gtkmm and glibmm. 

For gtkmm, the requirement is `FileChooserNative` the workaround used is to have a factory function which creates a file chooser dialog and returns `Glib::RefPtr<T>` where `T` is `Gtk::FileChooserNative` if supported and `Gtk::FileChooserDialog` if not (a warning is also issued with a link to #1000). 

For glibmm the requirement is `canonicalize_filename` and the workaround is to use `std::filesystem`. If `std::filesystem` is not supported then it fails with an error linking to #918 or suggesting that they may be using llvm < 9. The only problem here is `fs::canonical` _must_ be used if compiling on older systems is to be supported, I have refracted the existing code to support this.

I also reduced the cmake minimum required version to 3.1, in order to do this I had to move the `install()` for the `cherrytree` target into `/src/ct/CMakeLists.txt`. I tested installing and it installed to the correct directories.

With these changes I successfully compiled on ubuntu 18 with gtkmm 3.22.2, glibmm 2.56.0, cmake 3.10.2, and llvm9. I also tested it on my fedora 32 system which built it and ran without issue. 

Resolves:  #1035

### CODE REVIEW

File: future/src/ct/ct_filesystem.cc [hunk 1]
```diff
@@ -33,6 +33,24 @@
 #include "ct_logging.h"
 #include "ct_config.h"
 
+#if GLIBMM_MAJOR_VERSION <= 2 && GLIBMM_MINOR_VERSION <= 64 // canonicalize_filename is since 2.64
```
Comment (txe) [inline future/src/ct/ct_filesystem.cc:36]: maybe GLIBMM_MINOR_VERSION < 64

Comment (ForeverRainbow) [reply future/src/ct/ct_filesystem.cc:36]: oops lol, yep that should be strict

**Comment by giuspen** [comment]
looks good

**Comment by giuspen** [comment]
@ForeverRainbow the file chooser dialog is broken in Ubuntu 20.04 after this, could you have a look? It's easy to reproduce just try to file--open a cherrytree document you won't be able to do it

**Comment by giuspen** [comment]
@ForeverRainbow also windows/msys2 has the same issue, I assume you only tested the 18.04 build now worked but you broke everything else :P

**Comment by ForeverRainbow** [comment]
I'll check it out when i get home lol, oops. How is the file chooser "broken"? Is it just not using the native one?

**Comment by ForeverRainbow** [comment]
@giuspen cannot reproduce this on fedora. I will try windows later but its strange because (due to a macro mistake) the Gtk: :FileChooserDialog is always used currently and it was used up until recently when txe changed it. 

I have fixed the macro mistake and will make a pr later but would like to include a fix for this in it first

**Comment by giuspen** [comment]
It must have been something wrong with the smart pointers that you introduced. This of supporting an older system build must not add complication, in fact I would prefer if you avoid pointers at all

**Comment by ForeverRainbow** [comment]
@giuspen Fair enough, do you want me to try and fix this or just leave it? (I cannot remove the pointers since `Gtk::FileChooserNative::create` returns a `Glib::RefPtr<...>`)

**Comment by giuspen** [comment]
@txe what do you think about this?

**Comment by ForeverRainbow** [comment]
Ok I have just reproduced this on windows. Cannot select any file to open doubling clicking does nothing

**Comment by ForeverRainbow** [comment]
@giuspen Ok so using the native dialog does fix this. Looking at the history `Gtk::FileChooserDialog` was previousely locally allocated so I guess it doesnt play well with `Glib::RefPtr` (Its an ancient part of gtkmm afaik). Since msys will be using the latest gtkmm anyway it shouldn't ever need to use `FileChooserDialog`. 

I can make a pr with the fix and readding these changes if your ok with that?

**Comment by giuspen** [comment]
Agreed @ForeverRainbow let's keep it simple and drop the older systems, thanks for trying though

**Pull request merged**